applications/nrf5340_audio/src/bluetooth/bt_management/scanning/bt_mgmt_scan_for_broadcast.c

```diff
@@ -256,6 +257,149 @@ static struct bt_le_per_adv_sync_cb sync_callbacks = {
+		}
+	}
+}
+
+static K_WORK_DELAYABLE_DEFINE(pa_timer, pa_timer_handler);
+
+/**
+ * @brief	Subscribe to periodic advertising sync transfer (PAST).
+ *
+ * @param[in]	conn	Pointer to the connection object.
+ * @param[in]	pa_interval	Periodic advertising interval.
+ * @return	0 if success, error otherwise.
+ */
+static int pa_sync_past(struct bt_conn *conn, uint16_t pa_interval)
+{
+	int ret;
+	struct bt_le_per_adv_sync_transfer_param param = {0};
+
+	param.skip = PA_SYNC_SKIP;
+	param.timeout = interval_to_sync_timeout(pa_interval);
+
+	ret = bt_le_per_adv_sync_transfer_subscribe(conn, &param);
+	if (ret) {
+		LOG_WRN("Could not do PAST subscribe: %d", ret);
+		return ret;
+	}
+
+	LOG_DBG("Syncing with PAST: %d", ret);
+	(void)k_work_reschedule(&pa_timer, K_MSEC(param.timeout * 10));
+
+	return ret;
+}
+
+static int pa_sync_req_cb(struct bt_conn *conn,
+			  const struct bt_bap_scan_delegator_recv_state *recv_state,
+			  bool past_avail, uint16_t pa_interval)
+{
+	int ret;
+
+	req_recv_state = recv_state;
+
+	if (recv_state->pa_sync_state == BT_BAP_PA_STATE_SYNCED ||
+	    recv_state->pa_sync_state == BT_BAP_PA_STATE_INFO_REQ) {
+		LOG_DBG("Already syncing");
+		/* TODO: Terminate existing sync and then sync to new?*/
+		return -EALREADY;
+	}
+
+	LOG_INF("broadcast ID received = %X", recv_state->broadcast_id);
+	brcast_src_info.id = recv_state->broadcast_id;
+
+	if (IS_ENABLED(CONFIG_BT_PER_ADV_SYNC_TRANSFER_RECEIVER) && past_avail) {
+		ret = pa_sync_past(conn, pa_interval);
+		if (ret) {
+			LOG_WRN("Subscribe to PA sync PAST failed, ret = %d", ret);
+			return ret;
+		}
+
+		ret = bt_bap_scan_delegator_set_pa_state(req_recv_state->src_id,
+							 BT_BAP_PA_STATE_INFO_REQ);
+		if (ret) {
+			LOG_WRN("Set PA state to INFO_REQ failed, err = %d", ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static int pa_sync_term_req_cb(struct bt_conn *conn,
+			       const struct bt_bap_scan_delegator_recv_state *recv_state)
+{
+	int ret;
+	struct bt_mgmt_msg msg;
+
+	msg.event = BT_MGMT_BROADCAST_SINK_DISABLE;
+	ret = zbus_chan_pub(&bt_mgmt_chan, &msg, K_NO_WAIT);
+	ERR_CHK(ret);
+
+	return 0;
+}
+
+static void broadcast_code_cb(struct bt_conn *conn,
+			      const struct bt_bap_scan_delegator_recv_state *recv_state,
+			      const uint8_t broadcast_code[BT_AUDIO_BROADCAST_CODE_SIZE])
+{
+	int ret;
+	struct bt_mgmt_msg msg;
+
+	LOG_DBG("Broadcast code received for %p", (void *)recv_state);
+	memcpy(bt_mgmt_broadcast_code, broadcast_code, BT_AUDIO_BROADCAST_CODE_SIZE);
+
+	msg.event = BT_MGMT_BROADCAST_CODE_RECEIVED;
+	ret = zbus_chan_pub(&bt_mgmt_chan, &msg, K_NO_WAIT);
+	ERR_CHK(ret);
+}
+
+static int bis_sync_req_cb(struct bt_conn *conn,
+			   const struct bt_bap_scan_delegator_recv_state *recv_state,
+			   const uint32_t bis_sync_req[CONFIG_BT_BAP_BASS_MAX_SUBGROUPS])
+{
+	int ret;
+	struct bt_mgmt_msg msg;
+
+	/* Only support one BIG now */
+	LOG_DBG("BIS sync request received for %p: 0x%08x", (void *)recv_state, bis_sync_req[0]);
```

Comment:
Why is this hard coded to 0? The comment above makes it seem that this is BIG0, but this is subgroup0. We should probably traverse the bis_synq_req to see what BISes are set, it could be in bis_sync_req[1] potentially

Reply:
Yes, let's have a discussion for this part.

Reply:
As we discussed, there's no broadcast assistant from phone doesn't support subgroup assignment at this moment, so we  will lock with first subgroup at this moment.